src/ocamlorg_package/lib/opam_repository.ml

```diff
@@ -16,6 +16,15 @@ module Process = struct
 
   let check_status cmd = function
     | Unix.WEXITED 0 -> ()
+    | Unix.WEXITED 1 ->
+        if Pull.is_pull () then
+          prerr_endline
+            "\n\
+             Pulling from opam repository was not possible. \n\
+             We will continue by building the package state using the existing \
+             local state of opam-repository.\n\
+             Some functionality might not work properly.\n"
+        else Fmt.failwith "%a %a" pp_cmd cmd pp_status (Unix.WEXITED 1)
```

Comment:
I am conflicted by this.

The module `Process` seems to try to encapsulate calling arbitrary processes. So we can't really introduce anything specific to the pull request here.

A global reference (from a perspective of code health) should be a last resort strategy when there's no other, more local solution.

My impression is that `check_status` deciding that we `Fmt.failwith` is wrong in the first place.

The sensible thing for `check_status` to do would be to return a `Result` value. So it would either return `Ok` with the value we get in the successful case, or it would return `Error` with the formatted error message. Then the call sites of the `Process.exec`, `Process.pread`, `Process.pread_lines` decide, if an error is worth throwing an exception or not.

Reply:
Sorry for the late reply. I removed the global reference which was indeed bad design. I agree with your opinion about the `Process ` module but feel it is out of scope for this pull request.  I can open an issue about it and try to implement your suggestions if that is ok with you.